src/Analyzer.JsonRuleEngine/Expressions/LogicalExpression.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.Templates.Analyzer.Types;
+
+namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Expressions
+{
+    /// <summary>
+    /// Represents an anyOf expression in a JSON rule.
+    /// </summary>
+    internal class LogicalExpression : Expression
```

Comment:
I found the name a little confusing because most of the expressions in a way represent a logical property, right? Maybe it makes sense for this class to abstract list-based structured operators?

Reply:
Yeah that makes sense, I'll work on that.

Reply:
FYI, I was thinking logical [in this sense](https://www.sciencedirect.com/topics/engineering/logical-expression#:~:text=A%20logical%20expression%20is%20a,and%200%20to%20indicate%20false.) but maybe that's more of a "math" term. I like the list-based idea still.